localstack/services/sqs/provider.py

```diff
@@ -1168,6 +1169,30 @@ def _dead_letter_check(
         else:
             return False
 
+    def list_dead_letter_source_queues(
+        self,
+        context: RequestContext,
+        queue_url: String,
+        next_token: Token = None,
+        max_results: BoxedInteger = None,
+    ) -> ListDeadLetterSourceQueuesResult:
+        urls = []
+        backend = SqsBackend.get(context.region)
+        dead_letter_queue = self._resolve_queue(context, queue_url=queue_url)
+        for queue in backend.queues.values():
+
+            if queue.region != context.region:
+                continue
+            if queue.account_id != context.account_id:
+                continue
```

Comment:
we can omit these kinds of checks, since the sqs backend only contains queues of the respective region and context in the first place.

Reply:
remnant of the old backend, fixed :+1: